src/main/java/org/elasticsearch/action/index/IndexRequest.java

```diff
@@ -557,13 +559,22 @@ public boolean autoGeneratedId() {
         return this.autoGeneratedId;
     }
 
+    private static Version getIndexVersionFromIndex(String index, MetaData metaData) {
```

Comment:
Why not use `Version.indexeCreated`? How is it possible to be looking up created version for an index that doesn't exist?